packages/connect-react/src/index.ts

```diff
@@ -1,7 +1,139 @@
-export function useAragon(): void {
-  return
+import React, { useContext, useEffect, useMemo, useRef, useState } from 'react'
+import {
+  Application,
+  ConnectOptions,
+  ConnectorDeclaration,
+  Organization,
+  Permission,
+  connect,
+} from '@aragon/connect'
+
+type ContextValue = {
+  org: Organization | null
+  orgError: Error | null
+  orgLoading: boolean
+}
+
+type ConnectProps = {
+  children: React.ReactNode
+  connector: ConnectorDeclaration
+  location: string
+  options?: ConnectOptions
+}
+
+type OrganizationHookResult = [Organization | null, Error | null, boolean]
+type AppsHookResult = [Application[], Error | null, boolean]
+type PermissionsHookResult = [Permission[], Error | null, boolean]
+
+const ConnectContext = React.createContext<ContextValue>({
+  org: null,
+  orgError: null,
+  orgLoading: false,
+})
+
+export function Connect({
+  children,
+  connector,
+  location,
+  options,
+}: ConnectProps): JSX.Element {
+  const [org, setOrg] = useState<Organization | null>(null)
+  const [orgError, setOrgError] = useState<Error | null>(null)
+  const [orgLoading, setOrgLoading] = useState<boolean>(false)
+
+  useEffect(() => {
+    let cancelled = false
+
+    const update = async () => {
+      setOrgLoading(true)
+      try {
+        const org = await connect(location, connector, options)
+        if (!cancelled) {
+          setOrgError(null)
+          setOrg(org)
+        }
+      } catch (err) {
+        if (!cancelled) {
+          setOrgError(err)
+          setOrg(null)
+        }
+      } finally {
+        if (!cancelled) {
+          setOrgLoading(false)
+        }
+      }
+    }
+
+    update()
+
+    return () => {
+      cancelled = true
+    }
+  }, [location, connector, options])
+
+  const value = useMemo<ContextValue>(() => ({ org, orgError, orgLoading }), [
+    org,
+    orgError,
+    orgLoading,
+  ])
+
+  return React.createElement(ConnectContext.Provider, { value, children })
```

Comment:
Ooh interesting, why are we using `createElement` here? (Also, should we destructure it as we do with all the hooks?)

Reply:
> Ooh interesting, why are we using `createElement` here?

Just to avoid using JSX :-) This is the only element being created in this project, so I don’t think we really need JSX.

> (Also, should we destructure it as we do with all the hooks?)

I’m not sure I understand that one, could you please add a code example?

Reply:
My bad; I meant destructure `createElement`. :) As in

`import React, { createElement, ...} from 'react'`. Just for style!

Reply:
Oooh ok, yes let’s do this!